engines/sci/engine/kvideo.cpp

```diff
@@ -128,36 +128,35 @@ reg_t kShowMovie(EngineState *s, int argc, reg_t *argv) {
 	Common::ScopedPtr<Video::VideoDecoder> videoDecoder;
 
 	bool switchedGraphicsMode = false;
+	Graphics::PixelFormat screenFormat = g_system->getScreenFormat();
 
 	if (argv[0].isPointer()) {
 		Common::String filename = s->_segMan->getString(argv[0]);
 
 		if (g_sci->getPlatform() == Common::kPlatformMacintosh) {
 			// Mac QuickTime
 			// The only argument is the string for the video
+			videoDecoder.reset(new Video::QuickTimeDecoder());
 
-			// Switch to 16bpp graphics for Cinepak
-			if (g_system->getScreenFormat().bytesPerPixel == 1) {
-				const Common::List<Graphics::PixelFormat> supportedFormats = g_system->getSupportedFormats();
-				Common::List<Graphics::PixelFormat>::const_iterator it;
-				for (it = supportedFormats.begin(); it != supportedFormats.end(); ++it) {
-					if (it->bytesPerPixel == 2) {
-						const Graphics::PixelFormat format = *it;
-						initGraphics(screenWidth, screenHeight, &format);
-						switchedGraphicsMode = true;
-						break;
-					}
-				}
+			if (!videoDecoder->loadFile(filename)) {
+				error("Could not open '%s'", filename.c_str());
+				videoDecoder.reset();
 			}
 
-			if (g_system->getScreenFormat().bytesPerPixel == 1) {
-				warning("This video requires >8bpp color to be displayed, but could not switch to RGB color mode");
-				return NULL_REG;
+			// Handle the pixel format for Cinepak videos
+			if (videoDecoder && videoDecoder->getPixelFormat() != screenFormat) {
```

Comment:
At this point, the screen will be in 8bpp, but the video will be encoded in 16bpp, so this logic will always dither the video, instead of switching the screen to 16bpp

Reply:
The format that the screen is using is dependent on whether or not the "RGB rendering" option is enabled or not. If it's off, the screen will be 8bpp. If it's on, the screen will either be 16bpp or 32bpp depending on which one the backend prefers.